crates/cli/commands/src/stage/unwind.rs

```diff
@@ -60,54 +59,35 @@ impl<C: ChainSpecParser<ChainSpec: EthChainSpec + EthereumHardforks>> Command<C>
-                info!(target: "reth::cli", "Performing an unwind for offline-only data!");
-            }
-
-            if let Some(highest_static_file_block) = highest_static_file_block {
-                info!(target: "reth::cli", ?target, ?highest_static_file_block, "Executing a pipeline unwind.");
-            } else {
-                info!(target: "reth::cli", ?target, "Executing a pipeline unwind.");
-            }
-            info!(target: "reth::cli", prune_config=?config.prune, "Using prune settings");
-
-            // This will build an offline-only pipeline if the `offline` flag is enabled
-            let mut pipeline =
-                self.build_pipeline(config, provider_factory, components.evm_config().clone())?;
+        // Sanity check that we have static files available to unwind to the target block.
+        let highest_static_file_block =
+            provider_factory.static_file_provider().get_highest_static_files().max_block_num();
+        if highest_static_file_block
+            .filter(|highest_static_file_block| *highest_static_file_block > target)
+            .is_none()
+        {
+            return Err(eyre::eyre!("static files not available for target block {target}, highest is {highest_static_file_block:?}"));
+        }
 
-            // Move all applicable data from database to static files.
-            pipeline.move_to_static_files()?;
+        if self.offline {
+            info!(target: "reth::cli", "Performing an unwind for offline-only data!");
+        }
 
-            pipeline.unwind(target, None)?;
+        if let Some(highest_static_file_block) = highest_static_file_block {
+            info!(target: "reth::cli", ?target, ?highest_static_file_block, "Executing a pipeline unwind.");
         } else {
-            info!(target: "reth::cli", ?target, "Executing a database unwind.");
-            let provider = provider_factory.provider_rw()?;
+            info!(target: "reth::cli", ?target, "Executing a pipeline unwind.");
+        }
+        info!(target: "reth::cli", prune_config=?config.prune, "Using prune settings");
 
-            provider
-                .remove_block_and_execution_above(target)
-                .map_err(|err| eyre::eyre!("Transaction error on unwind: {err}"))?;
+        // This will build an offline-only pipeline if the `offline` flag is enabled
+        let mut pipeline =
+            self.build_pipeline(config, provider_factory, components.evm_config().clone())?;
 
-            // update finalized block if needed
-            let last_saved_finalized_block_number = provider.last_finalized_block_number()?;
-            if last_saved_finalized_block_number.is_none_or(|f| f > target) {
-                provider.save_finalized_block_number(target)?;
-            }
+        // Move all applicable data from database to static files.
+        pipeline.move_to_static_files()?;
 
-            provider.commit()?;
-        }
+        pipeline.unwind(target, None)?;
```

Comment:
is the reasoning behind this, that currently we pretty much always have static files and will be doing a pipeline unwind?

Reply:
Yes, with the work @shekhirin has been doing we always have static files, there's never a situation where static files are behind mdbx because we don't even write that data to mdbx anymore. So if static files don't have the data we can't do anything, just error out, otherwise always pipeline.